### PR TITLE
docs: clean up docs for azp migration

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -3,10 +3,10 @@
 # Bazel doesn't need more than 200MB of memory for local build based on memory profiling:
 # https://docs.bazel.build/versions/master/skylark/performance.html#memory-profiling
 # The default JVM max heapsize is 1/4 of physical memory up to 32GB which could be large
-# enough to consume all memory constrained by cgroup in large host, which is the case in CircleCI.
+# enough to consume all memory constrained by cgroup in large host.
 # Limiting JVM heapsize here to let it do GC more when approaching the limit to
 # leave room for compiler/linker.
-# The number 2G is choosed heuristically to both support in CircleCI and large enough for RBE.
+# The number 2G is choosed heuristically to both support in large VM and small VM with RBE.
```
Comment (phlax) [inline .bazelrc:9]: s/choosed/chosen/
s/suppport in large/support large/

File: ci/README.md [hunk 2]
```diff
@@ -189,10 +189,10 @@ This build the Ubuntu based `envoyproxy/envoy-build-ubuntu` image, and the final
 
 # macOS Build Flow
 
-The macOS CI build is part of the [CircleCI](https://circleci.com/gh/envoyproxy/envoy) workflow.
+The macOS CI build is part of the [Azure Pipelines](https://dev.azure.com/cncf/envoy/_build) workflow.
 Dependencies are installed by the `ci/mac_ci_setup.sh` script, via [Homebrew](https://brew.sh),
-which is pre-installed on the CircleCI macOS image. The dependencies are cached are re-installed
-on every build. The `ci/mac_ci_steps.sh` script executes the specific commands that
+which is pre-installed on the [Azure Pipelines macOS image](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md).
+The dependencies are cached are re-installed on every build. The `ci/mac_ci_steps.sh` script executes the specific commands that
```
Comment (phlax) [inline ci/README.md:195]: s/are cached are/are cached and/

**Comment by htuch** [comment]
LGTM modulo @phlax comments.

**Comment by lizan** [comment]
@htuch gently ping

**Review by htuch — APPROVED** [review]

**Pull request merged**